src/providers/arcgisrest/qgsarcgisrestdataitems.cpp

```diff
@@ -0,0 +1,674 @@
+    else
+      items.append( item );
+  }
+}
+
+QgsArcGisRestConnectionItem::QgsArcGisRestConnectionItem( QgsDataItem *parent, const QString &name, const QString &path, const QString &connectionName )
+  : QgsDataCollectionItem( parent, name, path, QStringLiteral( "AFS" ) )
+  , mConnName( connectionName )
+{
+  mIconName = QStringLiteral( "mIconConnect.svg" );
+  mCapabilities |= Collapse;
+
+  QgsSettings settings;
+  const QString key = QStringLiteral( "qgis/connections-arcgisfeatureserver/" ) + mConnName;
+  mPortalContentEndpoint = settings.value( key + "/content_endpoint" ).toString();
+  mPortalCommunityEndpoint = settings.value( key + "/community_endpoint" ).toString();
+}
+
+QVector<QgsDataItem *> QgsArcGisRestConnectionItem::createChildren()
+{
+  const QgsOwsConnection connection( QStringLiteral( "ARCGISFEATURESERVER" ), mConnName );
+  const QString url = connection.uri().param( QStringLiteral( "url" ) );
+  const QString authcfg = connection.uri().authConfigId();
+  const QString referer = connection.uri().param( QStringLiteral( "referer" ) );
+  QgsStringMap headers;
+  if ( ! referer.isEmpty() )
+    headers[ QStringLiteral( "Referer" )] = referer;
+
+  QVector<QgsDataItem *> items;
+  if ( !mPortalCommunityEndpoint.isEmpty() && !mPortalContentEndpoint.isEmpty() )
+  {
+    items << new QgsArcGisPortalGroupsItem( this, QStringLiteral( "groups" ), authcfg, headers, mPortalCommunityEndpoint, mPortalContentEndpoint );
+    items << new QgsArcGisRestServicesItem( this, url, QStringLiteral( "services" ), authcfg, headers );
+  }
+  else
+  {
+    QString errorTitle, errorMessage;
+    const QVariantMap serviceData = QgsArcGisRestQueryUtils::getServiceInfo( url, authcfg, errorTitle, errorMessage, headers );
+    if ( serviceData.isEmpty() )
+    {
+      if ( !errorMessage.isEmpty() )
+      {
+        std::unique_ptr< QgsErrorItem > error = qgis::make_unique< QgsErrorItem >( this, tr( "Connection failed: %1" ).arg( errorTitle ), mPath + "/error" );
+        error->setToolTip( errorMessage );
+        items.append( error.release() );
+        QgsDebugMsg( "Connection failed - " + errorMessage );
+      }
+      return items;
+    }
+
+    addFolderItems( items, serviceData, url, authcfg, headers, this, QString() );
+    addServiceItems( items, serviceData, url, authcfg, headers, this, QString() );
+    addLayerItems( items, serviceData, url, authcfg, headers, this, QgsArcGisRestQueryUtils::AllTypes, QString() );
+  }
+
+  return items;
+}
+
+bool QgsArcGisRestConnectionItem::equal( const QgsDataItem *other )
+{
+  const QgsArcGisRestConnectionItem *o = qobject_cast<const QgsArcGisRestConnectionItem *>( other );
+  return ( type() == other->type() && o && mPath == o->mPath && mName == o->mName );
+}
+
+QString QgsArcGisRestConnectionItem::url() const
+{
+  const QgsOwsConnection connection( QStringLiteral( "ARCGISFEATURESERVER" ), mConnName );
+  return connection.uri().param( QStringLiteral( "url" ) );
+}
+
+
+//
+// QgsAfsGroupsItem
```

Comment:
There seems to be a bunch of these section comments that are out of sync with the subsequent class name, e.g. QgsAfsGroupsItem vs QgsArcGisPortalGroupsItem.

Reply:
Fixed, thanks